rooms/views.py

```diff
@@ -223,3 +286,350 @@ def get(self, request, room_id):
+
+            if not is_updated:
+                return JsonResponse({'MESSAGE' : 'NOT EXISTS REVIEW'}, status=400)
+
+            return JsonResponse({'MESSAGE' : 'SUCCESS'}, status=200)
+
+        except ValueError:
+            return JsonResponse({'MESSAGE' : 'INVALID REQUEST'}, status=400)
+
+    @login_required
+    def patch(self, request):
+        try:
+            user_id   = request.user
+            datas     = json.loads(request.body)
+            review_id = int(datas['review_id'])
+            content   = datas['content']
+
+            is_updated = RoomReview.objects.filter(
+                user_id = user_id,
+                id      = review_id
+            ).update(
+                content = content
+            )
+
+            if not is_updated:
+                return JsonResponse({'MESSAGE' : 'NOT EXISTS REVIEW'}, status=400) 
+
+            return JsonResponse({'MESSAGE' : 'SUCCESS'}, status=200)
+
+        except KeyError:
+            return JsonResponse({'MESSAGE' : 'INVALID REQUEST'}, status=400)
+        except ValueError:
+            return JsonResponse({'MESSAGE' : 'INVALID REQUEST'}, status=400)
+        except json.decoder.JSONDecodeError:
+            return JsonResponse({'MESSAGE' : 'EMPTY BODY'}, status=400)
+
+class UserWish(View):
+    ### 위시리스트 ###
+    @login_required
+    def get(self, request):
+        user_id = request.user
+
+        wishes = Wish.objects.select_related('room') \
+                             .filter(user_id = user_id) \
+                             .prefetch_related('room__room_review', 'room__image_set')
+                            
+        result = [
+            {
+                'room_id'       : wish.room.id,
+                'price'         : wish.room.price,
+                'point_average' : FOMATTER(ReviewEvaluation.objects.filter(
+                                    review_id__in = wish.room.roomreview_set.values('id')
+                                ).aggregate(Avg('point'))['point__avg']),
+                'user_wish'     : 1,
+                'room_name'     : wish.room.name,
+                'room_image'    : [image.s3_url for image in wish.room.image_set.all()],
+                'review_count'  : wish.room.roomreview_set.all().aggregate(Count('id'))['id__count']
+            } for wish in wishes
+        ]
+
+        return JsonResponse({'wishes' : result}, status=200)
+
+    ### 위시 리스트 담기 ###
+    @login_required
+    def post(self, request):
+        try:
+            user_id = request.user
+            data    = json.loads(request.body)
+            room_id = data['room_id']
+
+            if not room_id:
+                return JsonResponse({'MESSAGE':'INVALID REQUEST'}, status=400)
+
+            obj, created = Wish.objects.get_or_create(
+                user_id = user_id,
+                room_id = data['room_id']
+            )
+
+            if not created:
+                return JsonResponse({'MESSAGE' : 'EXISTS WISH ROOM'}, status=400)
```

Comment:
단순히 중간테이블에 데이터가 있으면 생성하지 않고, 데이터가 없는 경우에는 새로 추가하는 로직에서는 아래와 같은 코드가 더 읽기 쉬운 것 같은데 어떠신가요?
`get_or_create`를 사용하나 아래와 같이 filter와 create을 각각 호출하나 쿼리 수는 차이가 없을 것 같습니다.

```suggestion
            if Wish.objects.filter( user_id = user_id, room_id = data['room_id']).exists():
                return JsonResponse({'MESSAGE' : 'EXISTS WISH ROOM'}, status=400)
            
            Wish.objects.create( user_id = user_id, room_id = data['room_id'])
            return return JsonResponse({'MESSAGE' : 'SUCCESS'}, status=201)
```